js/src/common/components/Select.tsx

```diff
@@ -0,0 +1,64 @@
+import Component, { ComponentProps } from "../Component";
+import icon from "../helpers/icon";
+
+export interface SelectProps extends ComponentProps {
+    /**u
+     * Whether or not the select is disabled.
+     */
+    disabled: boolean;
+
+    /**
+     * A callback to run when the selected value is changed.
+     */
+    onchange?: Function;
+
+    /**
+     * An object of options
+     */
+    options: {
+        [index: string]: string;
```

Comment:
Let's use `key` instead of `index` - index is targetted more for arrays, and the rest of instances where we typehint objects like this use `key`.

Reply:
yeah good idea